* Darwin and BSD need libgen.h for basename, on linux this header is
  also available

--- httpup.cpp
+++ httpup.cpp
@@ -15,6 +15,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <dirent.h>
+#include <libgen.h>
 
 #include "fileutils.h"
 #include "httpup.h"
